core/ir/instr_shared.c

```diff
@@ -100,6 +100,9 @@ instr_create(void *drcontext)
     is_instr_isa_mode_set = instr_set_isa_mode(instr, dr_get_isa_mode(dcontext));
 #endif
     CLIENT_ASSERT(is_instr_isa_mode_set, "setting instruction ISA mode unsuccessful");
+    if (!is_instr_isa_mode_set) {
```

Comment:
Instead, put `DODEBUG({});` around the lines related to is_instr_* (and move its decl down: it's up top b/c older Visual Studio required decls at top but none of our supported compilers require that anymore).

Reply:
instr_set_isa_mode() needs to be executed, so it needs to be placed outside of DODEBUG({}). I'm using IF_DEBUG() instead.